qa/task-cancellation/src/main/java/org/elasticsearch/plugin/tasks/BlockingCancellableTask.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.plugin.tasks;
+
+import org.apache.lucene.util.SetOnce;
+import org.elasticsearch.tasks.CancellableTask;
+import org.elasticsearch.tasks.TaskId;
+
+import java.util.Map;
+
+abstract class BlockingCancellableTask extends CancellableTask {
```

Comment:
why abstract?

Reply:
adjusted in https://github.com/elastic/elasticsearch/pull/59828/commits/c065a980791089b91b577d0bb9df17a2623d54ca